lightningd/peer_control.c

```diff
@@ -278,6 +362,63 @@ static void channel_config(struct lightningd *ld,
 	 ours->channel_reserve_satoshis = 0;
 };
 
+static void channel_errmsg(struct channel *channel,
+		      int peer_fd, int gossip_fd,
+		      const struct crypto_state *cs,
+		      u64 gossip_index,
+		      const struct channel_id *channel_id,
+		      const char *desc,
+		      const u8 *err_for_them)
+{
+	struct lightningd *ld = channel->peer->ld;
+	u8 *msg;
+
+	/* No peer fd means a subd crash or disconnection. */
+	if (peer_fd == -1) {
+		channel_fail_transient(channel, "%s: %s",
+				       channel->owner->name, desc);
+		return;
+	}
+
+	/* Do we have an error to send? */
+	if (err_for_them && !channel->error)
+		channel->error = tal_dup_arr(channel, u8,
+					     err_for_them,
+					     tal_len(err_for_them), 0);
+
+	/* BOLT #1:
+	 *
+	 * A sending node:
+	 *...
+	 *   - when `channel_id` is 0:
+	 *    - MUST fail all channels.
+	 *    - MUST close the connection.
+	 */
+	/* FIXME: Gossipd closes connection, but doesn't fail channels. */
+
+	/* BOLT #1:
+	 *
+	 * A sending node:
+	 *  - when sending `error`:
+	 *    - MUST fail the channel referred to by the error message.
+	 *...
+	 * The receiving node:
+	 *  - upon receiving `error`:
+	 *    - MUST fail the channel referred to by the error message.
+	 */
+	channel_fail_permanent(channel, "%s: %s ERROR %s",
+			       channel->owner->name,
+			       err_for_them ? "sent" : "received", desc);
```

Comment:
Is this `desc` the raw data from the `error` message? How do we know, this is safely printable to our log? Or, already filtered at this point?

Reply:
Indeed, already filtered.  desc comes from sanitize_error(), which does this work for us.  It also comes through wirestring, marshalling, which similarly fails on weird stuff.

Reply:
BTW, @ZmnSCPxj I sent mail to your protonmail, not sure how often you check it though...

Reply:
Not very often, but have seen and replied. Thank you very much your email.